cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -81,28 +82,60 @@ func (c *CStorPoolController) cStorPoolEventHandler(operation common.QueueOperat
 
 		// If pool is already present.
 		existingPool, _ := pool.GetPoolName()
-		if common.CheckIfPresent(existingPool, string(pool.PoolPrefix)+string(cStorPoolGot.GetUID())) {
+		flag := len(existingPool) != 0
+		// cnt is no of attempts to wait and handle in case of already present pool.
+		cnt := common.NoOfPoolWaitAttempts
+		for i := 0; flag && i < cnt; i++ {
 			common.InitialImportedPoolVol, err = volumereplica.GetVolumes()
-			if err != nil {
-				return string(apis.CStorPoolStatusOffline), err
+			// GetPoolName is to get pool name for particular no. of attempts.
+			existingPool, _ := pool.GetPoolName()
+			if common.CheckIfPresent(existingPool, string(pool.PoolPrefix)+string(cStorPoolGot.GetUID())) {
+				flag = true
+				// In the last attempt, ignore and update the status.
+				if i == cnt-1 {
+					if IsInitStatus(cStorPoolGot) {
+						glog.Infof("Pool %v is online", string(pool.PoolPrefix)+string(cStorPoolGot.GetUID()))
+						c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.AlreadyPresent), string(common.MessageResourceAlreadyPresent))
+						common.IsImported <- true
+						return string(apis.CStorPoolStatusOnline), nil
+					}
+					glog.Errorf("Pool %v already present", string(pool.PoolPrefix)+string(cStorPoolGot.GetUID()))
+					c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.AlreadyPresent), string(common.MessageResourceAlreadyPresent))
+					common.IsImported <- true
+					return string(apis.CStorPoolStatusAlreadyPresent), nil
+				}
+				glog.Infof("Attempt %v: Waiting...", i+1)
+				time.Sleep(common.PoolWaitInterval)
+			} else {
+				// If no pool is present while trying for getpoolname, set flag to false and
+				// break the loop, to import the pool later.
+				flag = false
 			}
-			glog.Errorf("Pool %v already present", string(pool.PoolPrefix)+string(cStorPoolGot.GetUID()))
-			c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.AlreadyPresent), string(common.MessageResourceAlreadyPresent))
-			return string(apis.CStorPoolStatusAlreadyPresent), nil
 		}
 
+		// import pool with cachefile.
```

Comment:
Lets have a single method to import pool with and without cache-file. The flow should be to first attempt to import the pool with the cache-file, if this fails attempt without the cache-file.
This can be taken up in the next PR.

Reply:
will address in next PR.